### DISCUSSION TITLE
Nuxt 3 catch-all page fills param with empty string instead of empty array

### DISCUSSION THREAD

**Discussion by hecktarzuli** [nuxt/nuxt#18178] [General]
When using the catch-all page file `/pages/abc/[...slug].vue`
- going to /abc/123 yields `slug:["123"]`
- going to /abc/ or /abc yields `slug:""`

I would have expected an empty array.  

cc:@pi0

**Comment by atinux** [comment]
Would love to have your input @posva

**Comment by posva** [comment]
Any optional parameter `?` or `*` can be empty and when it is, it's set to a falsy value so you can do this:

```js
if (route.params.slug) {
  // route.params.slug is an array with at least an item
}
```

and `[...slug]` is transformed to `:slug(.*)*`

**Reply by hecktarzuli** [reply]
I figured that was the reason, but wanted to bring it up anyway for a convo.  Thanks for your work!
